src/crimson/os/seastore/journal/circular_bounded_journal.cc

```diff
@@ -117,17 +117,15 @@ RecordScanner::read_validate_record_metadata_ret CircularBoundedJournal::read_va
 {
   LOG_PREFIX(CircularBoundedJournal::read_validate_record_metadata);
   paddr_t start = cursor.seq.offset;
-  return read_record(start, 0
+  return read_record(start, nonce
   ).safe_then([FNAME, &cursor](auto ret) {
     if (!ret.has_value()) {
       return read_validate_record_metadata_ret(
 	read_validate_record_metadata_ertr::ready_future_marker{},
 	std::nullopt);
     }
     auto [r_header, bl] = *ret;
-    if ((cursor.last_committed != JOURNAL_SEQ_NULL &&
-	cursor.last_committed > r_header.committed_to) ||
-	r_header.committed_to.segment_seq != cursor.seq.segment_seq) {
+    if (r_header.committed_to.segment_seq != cursor.seq.segment_seq) {
```

Comment:
I'm still not following why committed_to can define whether this record metadata is valid.

IIRC `committed_to` records the last fully-written seq record (must be a prior record that was committing) when starting to commit this record group. It should be irrelavent to whether this record group is valid.